java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -313,66 +325,133 @@ public void delete(final String rawPath) throws IOException {
     public void deleteRecursive(final String path) throws IOException {
         LOG.debug("DELETE {} [recursive]", path);
 
-        try {
-            this.delete(path);
-            LOG.debug("Finished deleting path {}", path);
-            return;
-        } catch (MantaClientHttpResponseException e) {
-            /* In the case of the directory not being empty, we let
-             * that error case go uncaught and pass through
-             * because it is in fact a directory with files. */
-            if (!e.getServerCode().equals(DIRECTORY_NOT_EMPTY_ERROR)) {
-                throw e;
-            }
-        }
-
-        try (Stream<MantaObject> objects = this.listObjects(path)) {
-            objects.forEach(obj -> {
-                try {
-                    if (obj.isDirectory()) {
-                        this.deleteRecursive(obj.getPath());
-                    } else {
-                        this.delete(obj.getPath());
+        /* We repetitively run the find() -> delete() stream operation and check
+         * the diretory targeted for deletion by attempting to delete it this
+         * deals with unpredictable directory contents changes and occasional
+         * delays in Manta where it will return from a file delete operation,
```

Comment:
Is this intended server side behavior?  If not, is there a MANTA issue?

Reply:
This is intermittent observed behavior. I honestly don't have a lot of time to go deep and do the reproduction steps needed to file an issue that can actually be acted upon. Past work has shown me that is a multi-day process.
I'm just going to delete the comment so that it doesn't lead anyone down a bad path.